### PR TITLE
modules/m_who: add fast path for WHO queries with exact nick

### PR DESCRIPTION
**Pull request by emersion**
When a WHO <nick> query is performed without a mask, use find_client
to immediately return a WHO reply without walking the global client
list.

Closes: https://github.com/oftc/oftc-hybrid/issues/64

### CODE REVIEW

**Comment by emersion** [comment]
Gentle ping @dwfreed @df7cb @Adam-

File: modules/m_who.c [hunk 1]
```diff
@@ -606,6 +634,11 @@ m_who(struct Client *client_p, struct Client *source_p, int parc, char *parv[])
       shown += do_who_channel(source_p, chan_p, 1);
     }
   }
+  else if(is_exact_nick_search())
+  {
+    if((target_p = find_client(wsopts.nick)) != NULL && IsClient(target_p))
```
Comment (Adam-) [inline modules/m_who.c:639]:
```suggestion
    if((target_p = find_person(client_p, wsopts.nick)) != NULL)
```

**Review by Adam- — COMMENTED** [review]
Not too sure why you pinged me but I am happy to review - note I am not affiliated with OFTC and also don't have merge privs

**Comment by emersion** [comment]
Thanks for the suggestion -- applied!

**Pull request merged**